alicloud/service_alicloud_cs.go

```diff
@@ -368,3 +392,90 @@ func (s *CsService) ignoreTag(t cs.Tag) bool {
 	}
 	return false
 }
+
+func (s *CsService) GetPermanentToken(clusterId string) (string, error) {
+
+	describeClusterTokensResponse, err := s.client.WithCsClient(func(csClient *cs.Client) (interface{}, error) {
+		return csClient.DescribeClusterTokens(clusterId)
+	})
+	if err != nil {
+		return "", fmt.Errorf("failed to get permanent token,because of %v", err)
+	}
+
+	tokens, ok := describeClusterTokensResponse.([]*cs.ClusterTokenResponse)
+
+	if ok != true {
+		return "", fmt.Errorf("failed to parse ClusterTokenResponse of cluster %s", clusterId)
+	}
+
+	permanentTokens := make([]string, 0)
+
+	for _, token := range tokens {
+		if token.Expired == 0 && token.IsActive == 1 {
+			permanentTokens = append(permanentTokens, token.Token)
+			break
+		}
+	}
+
+	// create a new token
+	if len(permanentTokens) == 0 {
+		createClusterTokenResponse, err := s.client.WithCsClient(func(csClient *cs.Client) (interface{}, error) {
+			clusterTokenReqeust := &cs.ClusterTokenReqeust{}
+			clusterTokenReqeust.IsPermanently = true
+			return csClient.CreateClusterToken(clusterId, clusterTokenReqeust)
+		})
+		if err != nil {
+			return "", fmt.Errorf("failed to create permanent token,because of %v", err)
```

Comment:
Please change line 428 to `return “”, WrapErrorf(err, DefaultErrorMsg, clusterId, “ CreateClusterToken”, AlibabaCloudSdkGoERROR) )`